heat/core/signal.py

```diff
@@ -77,24 +84,23 @@ def convolve(a: DNDarray, v: DNDarray, mode: str = "full") -> DNDarray:
     a = a.astype(promoted_type)
     v = v.astype(promoted_type)
 
-    if v.is_distributed():
-        raise TypeError("Distributed filter weights are not supported")
     if len(a.shape) != 1 or len(v.shape) != 1:
         raise ValueError("Only 1-dimensional input DNDarrays are allowed")
-    if a.shape[0] <= v.shape[0]:
-        raise ValueError("Filter size must not be greater than or equal to signal size")
     if mode == "same" and v.shape[0] % 2 == 0:
         raise ValueError("Mode 'same' cannot be used with even-sized kernel")
 
+    if v.shape[0] > a.shape[0]:
+        a, v = v, a
+
     # compute halo size
-    halo_size = v.shape[0] // 2
+    halo_size = int(v.lshape_map[0][0] / 2)
```

Comment:
This does not work. The halo size cannot depend on the local size of the first process, as this is arbitrary, e.g. 
one can set the local filter size at process 1 to zero with redistribute_, resulting in halo_size=0, regardless of the actual filter size.

Reply:
Basically, what I want is to have `halo_size` be the maximum local size of the `v`. I initially assumed `v` to be balanced. 

> local filter size at process 1 to zero with redistribute_

So, yes this won't work,I'll change `halo_size`  to be the maximum of all local chunk size.

My method works with balanced tensor on process, so if the distribution is uneven, I pad the local chunk with 0 till halo_size is reached. 

Will change this :+1:

Reply:
If the maximum local size will work you can do: 
halo_size = torch.max(v.lshape_map[:, 0]).item() // 2

Reply:
yes, done.